File: src/clang/magma/compile/rule/divide/DividingState.c
#include "DividingState.h"
DividingState append(char c);
int isLevel();
DividingState exit();
DividingState enter();
DividingState advance();
List_<String> segments();
int isShallow();
Option<Tuple<char, DividingState>> append();
Option<DividingState> appendAndDiscard();
Option<Tuple<char, DividingState>> pop();
Option<char> peek();

File: src/java/magma/compile/rule/divide/DividingState.java
package magma.compile.rule.divide;

import magma.collect.list.List_;
import magma.option.Option;
import magma.option.Tuple;

public interface DividingState {
    DividingState append(char c);

    boolean isLevel();

    DividingState exit();

    DividingState enter();

    DividingState advance();

    List_<String> segments();

    boolean isShallow();

    Option<Tuple<Character, DividingState>> popAndAppend();
}
